### PR TITLE
Base the damage that vehicles do to items they run over on their mass and wheel area.

### LINKED ISSUE #25261: Vehicles should "pulp" zombie corpses
This is a suggestion that impacts the collision code for
vehicles. That code is already due for refactoring (marked as a "big
TODO" in the code) but I'm still making this a separate issue as I
believe it can be implemented without too much trouble. 

**Is your feature request related to a problem? Please describe.**  

When zombies are run over by a vehicle, even at some speed, they are
left as "un-pulped" corpses. It seems strange that I can crush a zombie
with a tank but it will still get up after a while, whereas if i get
out and smash it with my fists, it is pulped beyond resurrection.

**Describe the solution you'd like**  
I would expect zombies struck by a vehicle at high speed to have a
certain percentage change of being pulped. The chance should increase
as a function of the weight and speed of the vehicle. I would suggest
computing the amount of energy absorbed by the zombie

`E = 1/2 * m * v^2`

This energy could then be multiplied by a modifier to account for
glancing blows or "lucky zombies". I would suggest the modifier take
the players driving skill and the speed (and toughness) of the zombie
into account.

The dead zombie would be pulped if the resulting "modified energy" is
above a suitable threshold.

If people agree that it is a good idea, I might take a stab at
implementing it.

### PR DESCRIPTION
**Pull request by Rhxydos**
#### Summary
```SUMMARY: Features "Make vehicles deal damage to items based on their mass and wheel area."```

#### Purpose of change
Currently, vehicles only deal 5 damage to objects they run over, which is only enough to damage items made out of materials with very low chip resistance, such as glass. That is both unrealistic and boring. This PR changes things so that running over objects with vehicles has the potential, particularly with heavy vehicles, to damage or destroy items outright. This is a double-edged sword, as while it can destroy potentially valuable items, it also allows for zombie corpses to be quickly disposed of when using something like a road roller. Fixes #25261.

#### Describe the solution
Vehicles use `map::smash_items()` to damage objects run over by vehicles, but before now, a static `power` value of 5 was fed into it, meaning it could only damage weak materials. This PR changes that so that the value is based on the weight that the vehicle is putting on the wheel that runs the items over.

To accomplish this, it adds an optional parameter to `map::vehicle_wheel_traction()` which causes it out output the area of the wheels that have contact with the ground. This was done to avoid code duplication. 

The calculations are done in `map::move_vehicle()`, because that is where `map::smash_items()` was already being called.  To convert the weight per wheel into something sensible, a value called `weight_to_damage_factor` is multiplied by the result of the calculation.

#### Describe alternatives you've considered
I considered using the total area of all wheels, regardless of whether or not they were touching the ground, but chose not to because I wanted to avoid rewriting existing code.

#### Additional context
Testing of this PR revealed several issues with using chip resistance for `map::smash_items()`. Even light vehicles like the bicycle can deal damage to giant ant corpses, liquids can be run over and destroyed, etc... Ideally, another value representing crushing resistance would be added to existing materials, but that is outside the scope of this PR. Another peculiarity is that corpses that are run over simply vanish, leaving behind a small puddle of blood. Changing this would also change what happens when corpses are shot or caught in an explosion, and while I do believe that they should spray blood and gore around in all 3 circumstances, I believe that it falls outside the scope of this PR due to it changing other systems.

Finally, `weight_to_damage_factor` is by no means set in stone, and can be tweaked if need be. I simply picked a value that resulted in what I felt were reasonable results.

### CODE REVIEW

File: src/vehicle_move.cpp [hunk 1]
```diff
@@ -1477,6 +1478,12 @@ float map::vehicle_wheel_traction( const vehicle &veh ) const
                 break;
             }
         }
+
+        // Ignore the movement modifier if needed.
+        if( ignore_movement_modifiers == true ) {
```
Comment (AMurkin) [inline src/vehicle_move.cpp]:
```suggestion
        if( ignore_movement_modifiers ) {
```

**Comment by SageB** [comment]
Travis is failing due to taking too long. Is this somthing I need to figure out how to fix, or is it out of my hands?

**Comment by SageB** [comment]
Update: Now Travis is failing on `GCC 5.3 Make build with curses, Magiclysm, and no Localization` (line 1093.) with the message
`crafting_test.cpp:280: FAILED:
  CHECK( can_craft == expect_craftable )
with expansion:
  true == false`

Which is very strange considering I didn't touch anything about crafting, AFAIK. The unit tests run fine on my machine, so I don't know what's up.

File: src/map.cpp [hunk 2]
```diff
@@ -610,6 +610,14 @@ vehicle *map::move_vehicle( vehicle &veh, const tripoint &dp, const tileray &fac
...
     if( !vertical && can_move ) {
         const auto wheel_indices = veh.wheelcache; // Don't use a reference here, it causes a crash.
+
+        // Values to deal with crushing items.
+        // The math needs to be floating-point to work, so the values might as well be.
+        const float vehicle_grounded_wheel_area = static_cast<int>( vehicle_wheel_traction( veh, true ) );
+        const float weight_to_damage_factor = 0.05; // Nobody likes a magic number.
+        const float vehicle_mass_kg = to_kilogram( veh.total_mass() );
+
+
```
Comment (Night-Pryanik) [inline src/map.cpp]:
```suggestion
```

File: src/map.h [hunk 3]
```diff
@@ -571,8 +571,9 @@ class map
         bool displace_water( const tripoint &dp );
 
         // Returns the wheel area of the vehicle multiplied by traction of the surface
+        // When ignore_movement_modifiers is set to true, it returns the area of the wheels touching the ground
         // TODO: Remove the ugly sinking vehicle hack
-        float vehicle_wheel_traction( const vehicle &veh ) const;
+        float vehicle_wheel_traction( const vehicle &veh,  bool ignore_movement_modifiers = false ) const;
```
Comment (Night-Pryanik) [inline src/map.h]:
```suggestion
        float vehicle_wheel_traction( const vehicle &veh, bool ignore_movement_modifiers = false ) const;
```

**Comment by SageB** [comment]
Oh jeez, I didn't realize all those rebased commits would show up. I'm just gonna close this while I try to fix this.

**Pull request closed without merging**